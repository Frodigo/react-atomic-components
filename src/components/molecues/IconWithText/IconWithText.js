import React from 'react';
import PropTypes from 'prop-types';

import classes from './IconWithText.module.css';

const IconWithText = (props) => {
    const { image, label, isActive, ...rest } = props;

    if (!image) return null;

    const containerClass = isActive
        ? classes.iconContainerActive
        : classes.iconContainer;
    const imageElement = isActive ? (
        <img
            className={classes.activeIcon}
            src={image}
            alt={label}
        />
    ) : (
        <img
            src={image}
            alt={label}
        />
    );

    return (
        <span className={containerClass} {...rest}>
            {imageElement}
            <span className={classes.label}>{label}</span>
        </span>
    );
};

IconWithText.propTypes = {
    image: PropTypes.shape({
        alt: PropTypes.string,
        url: PropTypes.string
    }).isRequired,
    isActive: PropTypes.bool,
    label: PropTypes.string.isRequired
};

IconWithText.defaultProps = {
    isActive: false
};

export default IconWithText;
