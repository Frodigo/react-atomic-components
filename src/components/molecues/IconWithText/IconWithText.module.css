.iconContainer {
    align-items: center;
    color: var(--color-gray-dark-3);
    cursor: pointer;
    display: flex;
    height: 55px;
    font-size: 12px;
    font-weight: 400;
    justify-content: flex-end;
    flex-direction: column;
    text-align: center;
    text-decoration: none;
    min-width: 50px;
}

.iconContainerActive {
    composes: iconContainer;
    background-color: var(--color-primary);
    color: var(--color-white);

}

.label {
    margin: 4px 0 2px;
}

.activeIcon path {
    fill: var(--color-white);
}
