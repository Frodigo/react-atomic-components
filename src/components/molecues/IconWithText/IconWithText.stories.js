import React from 'react';

//import Icon from '../../../public/assets/icons/home.svg';

import { default as IconWithTextComponent } from './IconWithText';

export default {
    title: 'Molecues/IconsWithText',
    component: IconWithTextComponent
};

const Template = (args) => (
    <div style={{ backgroundColor: '#000' }}>
        <IconWithTextComponent {...args} />
    </div>
);

export const Home = Template.bind({});
Home.args = {
    image: 'https://via.placeholder.com/20x20',
    label: 'Home'
};
