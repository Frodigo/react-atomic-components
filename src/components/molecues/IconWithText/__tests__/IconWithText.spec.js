import React from 'react';
import { render } from '@testing-library/react';

import IconWithText from '../IconWithText';

describe('IconWithText should be rendered', () => {
    const headingMock = 'Home';

    test('IconWithText renders correctly', () => {
        const { container } = render(
            <IconWithText label={headingMock} image={{alt: 'figure', url: 'https://via.placeholder.com/400x600'}} />
        );

        expect(container).toMatchSnapshot();
    });
});
