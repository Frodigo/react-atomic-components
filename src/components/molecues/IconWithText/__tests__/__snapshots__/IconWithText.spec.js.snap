// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconWithText should be rendered IconWithText renders correctly 1`] = `
<div>
  <span
    class="iconContainer"
  >
    <img
      alt="Home"
      src="[object Object]"
    />
    <span
      class="label"
    >
      Home
    </span>
  </span>
</div>
`;
